dynamic_env_pytest_tests/tests/services/s3_gate/test_s3_gate.py

```diff
@@ -0,0 +1,559 @@
+                )
+                if got_attrs:
+                    assert (
+                        got_attrs.get("VersionId") == version_id
+                    ), f"Expected VersionId is {version_id}"
+
+        with allure.step("Delete object and check it was deleted"):
+            response = s3_gate_object.delete_object_s3(self.s3_client, bucket, obj_key)
+            version_id_delete = response.get("VersionId")
+
+            with pytest.raises(Exception, match=r".*Not Found.*"):
+                s3_gate_object.head_object_s3(self.s3_client, bucket, obj_key)
+
+        with allure.step("Get content for all versions and check it is correct"):
+            for version, content in (
+                (version_id_2, version_2_content),
+                (version_id_1, version_1_content),
+            ):
+                file_name = s3_gate_object.get_object_s3(
+                    self.s3_client, bucket, obj_key, version_id=version
+                )
+                got_content = get_file_content(file_name)
+                assert (
+                    got_content == content
+                ), f"Expected object content is\n{content}\nGot\n{got_content}"
+
+        with allure.step("Restore previous object version"):
+            s3_gate_object.delete_object_s3(
+                self.s3_client, bucket, obj_key, version_id=version_id_delete
+            )
+
+            file_name = s3_gate_object.get_object_s3(self.s3_client, bucket, obj_key)
+            got_content = get_file_content(file_name)
+            assert (
+                got_content == version_2_content
+            ), f"Expected object content is\n{version_2_content}\nGot\n{got_content}"
+
+    @pytest.mark.s3_gate_multipart
+    @allure.title("Test S3 Object Multipart API")
+    def test_s3_api_multipart(self, bucket, simple_object_size):
+        """
+        Test checks S3 Multipart API (Create multipart upload/Abort multipart upload/List multipart upload/
+        Upload part/List parts/Complete multipart upload).
+        """
+        parts_count = 3
+        file_name_large = generate_file(
+            simple_object_size * 1024 * 6 * parts_count
+        )  # 5Mb - min part
+        object_key = self.object_key_from_file_path(file_name_large)
+        part_files = split_file(file_name_large, parts_count)
+        parts = []
+
+        uploads = s3_gate_object.list_multipart_uploads_s3(self.s3_client, bucket)
+        assert not uploads, f"Expected there is no uploads in bucket {bucket}"
+
+        with allure.step("Create and abort multipart upload"):
+            upload_id = s3_gate_object.create_multipart_upload_s3(
+                self.s3_client, bucket, object_key
+            )
+            uploads = s3_gate_object.list_multipart_uploads_s3(self.s3_client, bucket)
+            assert uploads, f"Expected there one upload in bucket {bucket}"
+            assert (
+                uploads[0].get("Key") == object_key
```

Comment:
not related to this exact test, but i suddenly became curious :thinking:

tested cluster is a single connected node system with a fixed synchronization model. It would be extremely useful to test the behavior of paired operations (like here `create->list`) accessing different segments of the cluster. I see this as a separate class of tests, but which is no less critical for cluster arch's

Reply:
https://github.com/nspcc-dev/neofs-testcases/issues/706